Index: content.module
===================================================================
--- content.module	(revision 893)
+++ content.module	(working copy)
@@ -149,7 +149,7 @@
 function content_load($node) {
   $cid = 'content:'. $node->nid .':'. $node->vid;
   if ($cached = cache_get($cid, 'cache_content')) {
-    return unserialize($cached->data);
+    return $cached->data;
   }
   else {
     $default_additions = _content_field_invoke_default('load', $node);
@@ -164,7 +164,7 @@
         $default_additions[$key] = $value;
       }
     }
-    cache_set($cid, 'cache_content', serialize($default_additions));
+    cache_set($cid, 'cache_content', $default_additions);
     return $default_additions;
   }
 }
@@ -725,7 +725,7 @@
 
   if ($reset || !isset($info)) {
     if ($cached = cache_get('content_type_info', 'cache_content')) {
-      $info = unserialize($cached->data);
+      $info = $cached->data;
     }
     else {
       $info = array(
@@ -806,7 +806,7 @@
         $info['content types'][$type['type']] = $type;
       }
 
-      cache_set('content_type_info', 'cache_content', serialize($info));
+      cache_set('content_type_info', 'cache_content', $info);
     }
   }
   return $info;
Index: fieldgroup.module
===================================================================
--- fieldgroup.module	(revision 893)
+++ fieldgroup.module	(working copy)
@@ -234,7 +234,7 @@
 
   if (!isset($groups) || $reset) {
     if ($cached = cache_get('fieldgroup_data', 'cache_content')) {
-      $data = unserialize($cached->data);
+      $data = $cached->data;
       $groups = $data['groups'];
       $groups_sorted = $data['groups_sorted'];
     }
@@ -256,7 +256,7 @@
       while ($field = db_fetch_array($result)) {
         $groups[$field['type_name']][$field['group_name']]['fields'][$field['field_name']] = $field;
       }
-      cache_set('fieldgroup_data', 'cache_content', serialize(array('groups' => $groups, 'groups_sorted' => $groups_sorted)));
+      cache_set('fieldgroup_data', 'cache_content', array('groups' => $groups, 'groups_sorted' => $groups_sorted));
     }
   }
 
