? gallerix-DRUPAL-5--1-2.patch
Index: gallerix.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/gallerix/gallerix.module,v
retrieving revision 1.6
diff -u -p -r1.6 gallerix.module
--- gallerix.module	28 Feb 2008 04:28:28 -0000	1.6
+++ gallerix.module	27 Jul 2008 13:18:56 -0000
@@ -259,7 +259,7 @@ function gallerix_fetch_picture($pid) {
   //Return cached data, if any.
   if ($cache && $picture = cache_get($pid, 'cache_gallerix')) {
     print "From Cache \n";
-    print gallerix_to_js(unserialize($picture->data));
+    print gallerix_to_js($picture->data);
     exit();
   }  
 
@@ -296,7 +296,7 @@ function gallerix_fetch_picture($pid) {
   
   //Cache all the picture information.
   if ($cache) {
-    cache_set($pid, 'cache_gallerix', serialize($data), CACHE_TEMPORARY);
+    cache_set($pid, 'cache_gallerix', $data, CACHE_TEMPORARY);
   }
   
   print gallerix_to_js($data);
