? gmap-DRUPAL-5--1-0-BETA3.patch
Index: gmap.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/gmap/gmap.module,v
retrieving revision 1.51.2.141
diff -u -p -r1.51.2.141 gmap.module
--- gmap.module	15 Jul 2008 15:45:25 -0000	1.51.2.141
+++ gmap.module	27 Jul 2008 13:25:26 -0000
@@ -1228,14 +1228,14 @@ function gmap_get_marker_titles($reset =
 
   $titles = cache_get('gmap_marker_titles');
   if ($titles) {
-    $titles = unserialize($titles->data);
+    $titles = $titles->data;
   }
 
   if ($reset || !$titles) {
     require_once(drupal_get_path('module', 'gmap') .'/gmap_markerinfo.inc');
     $titles = _gmap_get_marker_titles();
   }
-  cache_set('gmap_marker_titles', 'cache', serialize($titles));
+  cache_set('gmap_marker_titles', 'cache', $titles);
   return $titles;
 }
 
@@ -1250,14 +1250,14 @@ function gmap_get_icondata($reset = FALS
 
   $icons = cache_get('gmap_icondata');
   if ($icons) {
-    $icons = unserialize($icons->data);
+    $icons = $icons->data;
   }
 
   if ($reset || !$icons) {
     require_once(drupal_get_path('module', 'gmap') .'/gmap_markerinfo.inc');
     $icons = _gmap_get_icondata();
   }
-  cache_set('gmap_icondata', 'cache', serialize($icons));
+  cache_set('gmap_icondata', 'cache', $icons);
   return $icons;
 }
 
