Index: user_relationships_api.inc
===================================================================
--- user_relationships_api.inc	(revision 1624)
+++ user_relationships_api.inc	(working copy)
@@ -79,7 +79,7 @@
   if (!key_exists($uid, $relationships)) {
     $relationships[$uid] = array();
     if (($cache = cache_get("user_relationships_relationships_{$uid}", 'cache_user_relationships')) && !empty($cache->data)) {
-      $relationships[$uid] = unserialize($cache->data);
+      $relationships[$uid] = $cache->data;
     }
     else {
       $results = db_query(
@@ -92,7 +92,7 @@
         $relationships[$uid][] = $relationship;
       }
       if ($cache_throttle >= 0 && sizeof($relationships[$uid]) < $cache_throttle) {
-        cache_set("user_relationships_relationships_{$uid}", 'cache_user_relationships', serialize($relationships[$uid]));
+        cache_set("user_relationships_relationships_{$uid}", 'cache_user_relationships', $relationships[$uid]);
       }
     }
   }
@@ -115,7 +115,7 @@
 
   if (!$user_relationships[$rid]) {
     if (($cache = cache_get("user_relationships_relationship_{$rid}", 'cache_user_relationships')) && !empty($cache->data)) {
-      $user_relationships[$rid] = unserialize($cache->data);
+      $user_relationships[$rid] = $cache->data;
     }
     elseif ($relationship = db_fetch_object(db_query("SELECT * FROM {user_relationships} WHERE rid = %d", $rid))) {
       $relationship->requester    = user_load(array('uid' => $relationship->requester_id));
@@ -126,7 +126,7 @@
 
       _user_relationships_invoke('load', $relationship);
       $user_relationships[$rid] = $relationship;
-      cache_set("user_relationships_relationship_{$rid}", 'cache_user_relationships', serialize($relationship));
+      cache_set("user_relationships_relationship_{$rid}", 'cache_user_relationships', $relationship);
     }
   }
 
