? views-DRUPAL-5--1-6.patch
Index: views.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/views/views.module,v
retrieving revision 1.166.2.43
diff -u -p -w -r1.166.2.43 views.module
--- views.module	14 Jul 2007 18:54:16 -0000	1.166.2.43
+++ views.module	27 Jul 2008 13:21:19 -0000
@@ -200,10 +200,10 @@ function views_get_all_urls() {
           $views[$view->name] = $view->url;
       }
     }
-    cache_set("views_urls", 'cache_views', serialize($views));
+    cache_set("views_urls", 'cache_views', $views);
   }
   else {
-    $views = unserialize($cache->data);
+    $views = $cache->data;
   }
 
   return $views;
Index: views_cache.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/views/Attic/views_cache.inc,v
retrieving revision 1.2.2.18
diff -u -p -w -r1.2.2.18 views_cache.inc
--- views_cache.inc	14 Jul 2007 19:12:02 -0000	1.2.2.18
+++ views_cache.inc	27 Jul 2008 13:21:19 -0000
@@ -28,7 +28,7 @@ function _views_get_arguments($titles = 
 
   if (!$views_arguments) {
     $data = cache_get("views_arguments:$locale", 'cache_views');
-    $cache = unserialize($data->data);
+    $cache = $data->data;
     if (is_array($cache)) {
       $views_arguments = $cache;
     }
@@ -54,7 +54,7 @@ function _views_get_arguments($titles = 
         $views_arguments['title'][$name] = $arg;
       }
       $cache = $views_arguments;
-      cache_set("views_arguments:$locale", 'cache_views', serialize($cache));
+      cache_set("views_arguments:$locale", 'cache_views', $cache);
     }
   }
   return ($titles ? $views_arguments['base'] : $views_arguments['title']);
@@ -70,7 +70,7 @@ function _views_get_tables($full = false
 
   if (!$views_tables) {
     $data = cache_get("views_tables:$locale", 'cache_views');
-    $cache = unserialize($data->data);
+    $cache = $data->data;
 
     if (is_array($cache)) {
       $views_tables = $cache;
@@ -143,7 +143,7 @@ function _views_get_tables($full = false
           }
         }
       }
-      cache_set("views_tables:$locale", 'cache_views', serialize($views_tables));
+      cache_set("views_tables:$locale", 'cache_views', $views_tables);
     }
   }
   return ($full ? $views_tables : $views_tables['tables']);
@@ -224,7 +224,7 @@ function _views_get_default_views() {
 
   if (!$views_default_views) {
     $data = cache_get("views_default_views:$locale", 'cache_views');
-    $cache = unserialize($data->data);
+    $cache = $data->data;
 
     if (is_array($cache)) {
       $views_default_views = $cache;
@@ -243,7 +243,7 @@ function _views_get_default_views() {
           $views_default_views[$i] = $view;
         }
       }
-      cache_set("views_default_views:$locale", 'cache_views', serialize($views_default_views));
+      cache_set("views_default_views:$locale", 'cache_views', $views_default_views);
     }
   }
   return $views_default_views;
@@ -266,7 +266,7 @@ function _views_sort_arrays($a, $b) {
 
 function _views_get_query(&$view, $args, $filters) {
   if ($view->is_cacheable && ($cached = cache_get('views_query:' . $view->name, 'cache_views'))) {
-    $info = unserialize($cached->data);
+    $info = $cached->data;
 
     $plugins = _views_get_style_plugins();
     if ($plugins[$view->type]['needs_table_header']) {
@@ -281,7 +281,7 @@ function _views_get_query(&$view, $args,
         'countquery' => _views_replace_args($info['countquery'], $info['args']),
     );
     if ($view->is_cacheable) {
-      cache_set('views_query:' . $view->name, 'cache_views', serialize($data));
+      cache_set('views_query:' . $view->name, 'cache_views', $data);
     }
     $info = array_merge($info, $data);
   }
@@ -313,7 +313,7 @@ function _views_get_style_plugins($title
 
   if (!$views_style_plugins) {
     $data = cache_get("views_style_plugins:$locale", 'cache_views');
-    $cache = unserialize($data->data);
+    $cache = $data->data;
     if (is_array($cache)) {
       $views_style_plugins = $cache;
     }
@@ -328,7 +328,7 @@ function _views_get_style_plugins($title
         $views_style_plugins['base'][$name] = $arg;
       }
       $cache = $views_style_plugins;
-      cache_set("views_style_plugins:$locale", 'cache_views', serialize($cache));
+      cache_set("views_style_plugins:$locale", 'cache_views', $cache);
     }
   }
   return ($titles ? $views_style_plugins['title'] : $views_style_plugins['base']);
