Index: includes/bootstrap.inc
===================================================================
RCS file: /cvs/drupal/drupal/includes/bootstrap.inc,v
retrieving revision 1.144
diff -u -F^f -r1.144 bootstrap.inc
--- includes/bootstrap.inc	2 Jan 2007 10:21:39 -0000	1.144
+++ includes/bootstrap.inc	11 Jan 2007 18:36:48 -0000
@@ -806,6 +806,7 @@ function _drupal_bootstrap($phase) {
       break;
 
     case DRUPAL_BOOTSTRAP_SESSION:
+      require_once './includes/memcache.inc';
       require_once './includes/session.inc';
       session_set_save_handler('sess_open', 'sess_close', 'sess_read', 'sess_write', 'sess_destroy_sid', 'sess_gc');
       session_start();
@@ -875,6 +876,7 @@ function _drupal_cache_init($phase) {
  */
 function drupal_maintenance_theme() {
   global $theme;
+  require_once './includes/memcache.inc';
   require_once './includes/path.inc';
   require_once './includes/theme.inc';
   require_once './includes/common.inc';
