Index: modules/user/user.module
===================================================================
RCS file: /cvs/drupal/drupal/modules/user/user.module,v
retrieving revision 1.745
diff -u -r1.745 user.module
--- modules/user/user.module	11 Jan 2007 08:52:45 -0000	1.745
+++ modules/user/user.module	13 Jan 2007 18:43:48 -0000
@@ -82,6 +82,8 @@
       $user->roles[$role->rid] = $role->name;
     }
     user_module_invoke('load', $array, $user);
+
+    $user->access = dmemcache_get($user->uid, 'user_access');
   }
   else {
     $user = FALSE;
